WordPress/gutenberg
# Tag Cloud Block: Prevent block support styles and additional CSS classes from being printed twice

Follow-up on #43367
Similar to #44438
Blocker: ##44491

_Note_: I think the double spacing is a major layout issue, so it would be a good idea to be backported to WordPress 6.1.

_@edit_: As per the discussion that began with [this comment](https://github.com/WordPress/gutenberg/pull/44438#pullrequestreview-1119477064), it was decided to move the logic for filtering the block support attributes to the `ServerSideRender` component itself. Once the appropriate prop is added to the `ServerSideReder` component, the code changes in this PR will be simpler. This PR will be placed on freeze until a PR regarding improvements to `ServerSideRender` is submitted and merged.

_@update_: Now that #44491 has been merged, the new `skipBlockSupportAttributes ` prop of the `ServerSideRender` component is available. This allows us to solve the problem this PR addresses in a simpler way.

## What?
This PR fixes a problem in the editor where block support inline styles and additional CSS classes are rendered twice.

## Why?
This is because this block uses the `ServiceSideRender` component and receives all attributes (including styles generated by the block support).

## How?
As the discussion started with [this comment](https://github.com/WordPress/gutenberg/issues/43243#issuecomment-1229114714), there are many possible approaches for a block using the `ServiceSideRender` component. I have tried to solve this by not passing block support styles and additional classes to the `ServiceSideRender` component.

## Testing Instructions
- Add a tag cloud block.
- Apply some block support styles (Typography, Dimensions).
- Add additional CSS class.
- Confirm that the block support styles and additional CSS classes aren't applied to the `ServiceSideRender` component and are rendered only in the block wrapper by using the browser console.

## Screenshots or screencast 

### Before

![before](https://user-images.githubusercontent.com/54422211/192127914-a8c58186-1286-4a38-948f-b1fa0e604efd.png)

### After

![after](https://user-images.githubusercontent.com/54422211/192127915-7c2165d9-63a4-46a0-a32c-a29808864ef3.png)
